lib/Rex/Commands/Mkfs.pm

```diff
@@ -0,0 +1,75 @@
+
+=head1 NAME
+
+Rex::Commands::Mkfs - Create filesystems
+
+=head1 DESCRIPTION
+
+With this module you can create filesystems on existing partitions and logical volumes.
+
+=head1 SYNOPSIS
+
+ use Rex::Commands::Mkfs;
+
+=head1 EXPORTED FUNCTIONS
+
+=cut
+
+package Rex::Commands::Mkfs;
+
+use warnings;
+use strict;
+
+# VERSION
+
+require Rex::Exporter;
+use base qw(Rex::Exporter);
+use vars qw(@EXPORT);
+
+@EXPORT = qw(mkfs);
+
+use Rex::Commands::Run;
+use Carp;
+
+=head2 mkfs($devname, %option)
+
+Create a filesystem on device $devname.
+
+ mkfs "sda1",
+   fstype => "ext2",
+   label  => "mydisk";
+
+ mkfs "sda2",
+   fstype => "swap";
+
+=cut
+
+sub mkfs {
+  my ( $devname, %option ) = @_;
+
+  if ( exists $option{fstype} ) {
+    if ( $option{fstype} eq "swap" ) {
+      Rex::Logger::info("Creating swap space on /dev/$devname");
+      run "mkswap -f /dev/$devname";
+    }
```

Comment:
I think it's not possible anymore to specify a label for a swap partition. Can you make sure this possibility is not lost?